todo/src/scripts/storageController.js

```diff
@@ -0,0 +1,87 @@
+class StorageController {
+    constructor() {
+        if (!localStorage.getItem('toDo')) {
+            localStorage.setItem('toDo', JSON.stringify({
+               'tasks' : {
+                    'o2go6gkfcrr': {
+                        id: 'o2go6gkfcrr',
+                        isCurrent: true,
+                        title: 'Make cool App',
+                        task: 'Make id sort',
+                        priority: 'Low',
+                        date: '1635494070101',
+                    }
+                },
+                'sortDirection': true
+            }
+            ));
+        }
+        this.data = JSON.parse(localStorage.getItem('toDo'));
+    }
+
+    updateStorage() {
+        localStorage.setItem('toDo', JSON.stringify(this.data));
+        this.data = JSON.parse(localStorage.getItem('toDo'));
+    }
+    
+    getItemsList() {
+        return this.separeteNewItems(Object.entries(this.data['tasks']).map(item => item[1]));
+    }
+
+    getItemData(id) {
+        return this.data.tasks[id];
+    }
+
+    separeteNewItems(data) {
+        const newItems = data.filter(item => item.isCurrent);
+        const oldItems = data.filter(item => !item.isCurrent);
+        return [newItems, oldItems];
+    }
+
+    sortItems(data, direction = true) {
+        
+    }
+
+    addNewItem(id, title, task, priority, date) {
+        this.data.tasks[id] = {
+            id,
+            title,
+            task,
+            priority,
+            date,
+            isCurrent: true,
+        }
```

Comment:
according to your code you update item by id, but you don't add it. Either logic or name is wrong.

Reply:
In the case of updating the task, ```id``` staны unchanged , access by ```id``` is indicated here ```Object.assign(this.data.tasks[id]```,  but I added id in update.
```suggestion
        Object.assign(this.data.tasks[id], {
            id,
            title,
            task,
            priority
        })
```